Switch from ssl.wrap_socket to SSLContext.wrap_socket

jheunis-bloomberg opened the pull request:
The python docs specify that since python 3.2, it is recommended to use `SSLContext.wrap_socket` instead of `ssl.wrap_socket` ("since it is both inefficient and has no support for SNI and hostname matching").

In python3.12, `ssl.wrap_socket` was removed, which causes the 100-continue test to reliably fail whenever it's configured with `is_secure=True`. Switching to the recommended `SSLContext.wrap_socket` (using the default SSL context) fixes this, allowing the test to run cleanly again.

jmundack commented:
@adamemerson  - thanks!
anything else needed here before merge?

adamemerson commented:
Schedule a test run against this branch on Teuthology, you'll need to make a custom branch of Ceph that points to this branch of s3-tests.

jacquesh committed:
Switch from ssl.wrap_socket to SSLContext.wrap_socket

The python docs specify that since python 3.2, it is recommended to use
`SSLContext.wrap_socket` instead of `ssl.wrap_socket` ("since it is both
inefficient and has no support for SNI and hostname matching").

In python3.12, `ssl.wrap_socket` was removed, which causes the 100-continue
test to reliably fail whenever it's configured with `is_secure=True`.
Switching to the recommended `SSLContext.wrap_socket` (using the default
SSL context) fixes this, allowing the test to run cleanly again.

Signed-off-by: Jacques Heunis <jheunis@bloomberg.net>

jheunis-bloomberg commented:
Independently fixed by Casey in https://github.com/ceph/s3-tests/pull/734 and ported along with the rest of the code to the ceph repo.
